src/nvim/misc1.c

```diff
@@ -1292,11 +1293,12 @@ int plines_win_nofold(win_T *wp, linenr_T lnum)
   width = wp->w_width - win_col_off(wp);
   if (width <= 0)
     return 32000;
-  if (col <= width)
+  if (col <= (unsigned int)width)
     return 1;
   col -= width;
   width += win_col_off2(wp);
-  return (col + (width - 1)) / width + 1;
+  assert(col <= INT_MAX);
```

Comment:
If the line length could really reach `INT32_MAX` and `col += 1` is used, we could have an overlfow for a not changing `width` of `1`:
`(INT_MAX+(1-1))/1+1 > INT_MAX`.